.github/workflows/main.yml

```diff
@@ -156,7 +156,20 @@ jobs:
         uses: seanmiddleditch/gha-setup-vsdevenv@v4
         with:
           arch: ${{ env.MODEL == '64' && 'x64' || 'x86' }}
+      - name: 'macOS arm64: Run arm64 smoke test'
+        if: matrix.os == 'macos-14'
+        run: |
+          set -uexo pipefail
+
+          cat >hello.d <<EOF
+          version (AArch64) {} else static assert(0);
+          extern(C) int puts(const(char)* s);
+          extern(C) int main() { puts("Hello world from native arm64!"); return 0; }
+          EOF
+
+          generated/osx/release/64/dmd -betterC -arm -run hello.d
```

Comment:
This would be the very first CI test of DMD compiling & linking a native arm64 executable. It currently fails to link; AFAICT, because the Mach-O object file still specifies the x86_64 architecture, although it contains arm64 code.

Reply:
I've tried to implement the trivial Mach-O header changes; now the linker complains about an invalid relocation:
```
ld: relocation at '_main'+0x0008 is not supported: r_address=0x8, r_type=1, r_extern=0, r_pcrel=1, r_length=2 in '/Users/runner/work/dmd/dmd/hello.o'
```

I'm not going down that rabbit hole. ;)